bootstrap/DedicatedBootstrapper.zig

```diff
@@ -0,0 +1,225 @@
+    imgui.sdl3.newFrame();
+    imgui.newFrame();
+
+    imgui.setNextWindowPos(.{ .x = 0, .y = 0 });
+    imgui.setNextWindowSize(imgui_ctx.getIo().DisplaySize);
+
+    self.selectionWindow();
+
+    imgui.render();
+    try renderer.clear();
+    try imgui.sdl_renderer.renderDrawData(imgui.getDrawData(), renderer);
+    try renderer.present();
+}
+
+fn selectionWindow(self: *DedicatedBootstrapper) void {
+    var open = true;
+    const draw = imgui.begin("Select Renderer", &open, imgui.c.ImGuiWindowFlags_NoDecoration | imgui.c.ImGuiWindowFlags_NoResize);
+    defer imgui.end();
+    if (!draw) return;
+
+    for (self.manifests, 0..) |manifest, i| {
+        _ = imgui.radioButton(manifest.value.name, &self.selected_manifest_idx, @intCast(i));
+    }
+
+    imgui.separator();
+    if (imgui.button("OK")) {
+        self.ui_shutdown = true;
+        self.bootstrap_shutdown = false;
+        self.ready_to_boot = true;
+    }
+    imgui.sameLine();
+    if (imgui.button("Cancel")) {
+        self.ui_shutdown = true;
+        self.bootstrap_shutdown = true;
+        self.ready_to_boot = false;
+    }
+}
+
+fn bootstrapEngine(self: *DedicatedBootstrapper, args: []const []const u8, gpa: std.mem.Allocator) void {
+    log.info(@src(), "Starting FrooxEngine in the background...", .{});
+    var child = renderite.Bootstrap.ChildInfo.init(args, gpa) catch @panic("Failed to start FrooxEngine");
+
+    var bootstrap = renderite.Bootstrap.initBootstrap(&child, gpa, copy, paste) catch @panic("Failed to bootstrap FrooxEngine");
+
+    log.info(@src(), "FrooxEngine spawned, waiting for user selection...", .{});
+    while (!self.ready_to_boot) {
+        if (self.bootstrap_shutdown) {
+            log.warn(@src(), "Bootstrapper is shutting down, won't boot", .{});
+            return;
+        }
+
+        std.Thread.sleep(100 * std.time.ns_per_ms);
+    }
+
+    log.info(@src(), "Ready to boot!", .{});
+
+    const manifest = self.manifests[@intCast(self.selected_manifest_idx)].value;
+    const executable = switch (builtin.os.tag) {
+        .windows => manifest.winExecutablePath,
+        else => unix_executable: {
+            if (manifest.runInWine orelse false) {
+                break :unix_executable manifest.winExecutablePath;
+            }
+
+            break :unix_executable manifest.unixExecutablePath;
+        },
+    };
+
+    log.info(@src(), "Starting renderer '{s}' at '{s}'", .{ manifest.name, executable });
+
+    var argv: bounded_array.BoundedArray([]const u8, 5) = .{};
+
+    argv.appendAssumeCapacity(executable);
+    argv.appendAssumeCapacity("-QueueName");
+    argv.appendAssumeCapacity(bootstrap.init_settings.queue_name.constSlice());
+    argv.appendAssumeCapacity("-QueueCapacity");
+
+    // SAFETY: it's big enough
+    var msg_buf: [32]u8 = undefined;
+    const msg = std.fmt.bufPrint(&msg_buf, "{d}", .{bootstrap.init_settings.queue_length}) catch unreachable;
+
+    argv.appendAssumeCapacity(msg);
```

Comment:
Something like this is much cleaner:

```suggestion
    var msg_buf: [32]u8 = undefined;    
    const argv: []const []const u8 = &.{
    	executable,
    	"-QueueName",
    	bootstrap.init_settings.queue_name.constSlice(),
    	"-QueueCapacity",
    	// SAFETY: it's big enough
    	std.fmt.bufPrint(&msg_buf, "{d}", .{bootstrap.init_settings.queue_length}) catch unreachable
    };
```

Reply:
In the future I will need to bring in more custom arguments, so this won't work.